plugins/parsers/csv/parser.go

```diff
@@ -257,8 +354,21 @@ outer:
 		}
 	}
 
+	// add metadata fields
+	for k, v := range p.MetadataTags {
+		_, exists := tags[k]
+		if exists {
+			continue
+		}
+		tags[k] = v
+	}
+
 	// add default tags
 	for k, v := range p.DefaultTags {
+		_, exists := tags[k]
+		if exists {
+			continue
+		}
```

Comment:
As discussed, we need to clarify if we really want to maintain this order of priorities...

Reply:
Keep as is (default overwrites row values) and handle the order in a separate PR as commented below.